Integration tests do not properly unset AWS_PROFILE

If `AWS_PROFILE` is set it can cause the integration tests `src/integration_tests` to fail. Specifically, tests that do not specify profile, and will only correctly run if the `default` profile is selected, fail when `AWS_PROFILE` is set to a profile other than `default`. The environment variable should be unset by the test framework.

```
$ export AWS_PROFILE=test
$ make integration-tests
...
configure.bats
 ✓ Login with unconfigured awscli
 ✓ Enable plugin in ~/.aws/config
docker-idp-login.bats
 ✓ Test --ecp-endpoint-url
 ✓ Login with Docker IdP
login.bats
 ✗ Login and logout
   (from function `assert_equal' in file tests/../test_helper/bats-assert/src/assert_equal.bash, line 40,
    in test file tests/login.bats, line 26)
     `assert_equal "$(<$AWS_SHARED_CREDENTIALS_FILE)" "$CREDS_AWS_FILE"' failed
   Configured for AWSCLI V1.
   ECP Endpoint URL [None]: Username [None]: Enable Keyring [False]: Duo Factor [None]: Role ARN [None]:
   -- values do not equal --
   expected (2 lines):
     [default]
     credential_process = aws-login --profile default
   actual (2 lines):
     [test]
     credential_process = aws-login --profile test
   --
   
 ✓ Ensure environment of plugin is the same as the standalone script
login_configure.bats
 ✗ Login with unconfigured default profile
   (from function `assert_output' in file tests/../test_helper/bats-assert/src/assert_output.bash, line 194,
    in test file tests/login_configure.bats, line 16)
     `assert_output "$ERROR_MESG"' failed
   Configured for AWSCLI V1.
   
   -- output differs --
   expected (4 lines):
     Credential process is not set for current profile "default".
     Reconfigure using:
   
     aws login configure
   actual (4 lines):
     Credential process is not set for current profile "test".
     Reconfigure using:
   
     aws login configure
   --
   
 ✗ Configure nonexistent default profile
   (from function `assert_equal' in file tests/../test_helper/bats-assert/src/assert_equal.bash, line 40,
    from function `test_default_profile' in file tests/login_configure.bats, line 43,
    in test file tests/login_configure.bats, line 21)
     `test_default_profile' failed
   Configured for AWSCLI V1.
   
   -- values do not equal --
   expected (4 lines):
     [default]
     ecp_endpoint_url = https://shibboleth.illinois.edu/idp/profile/SAML2/SOAP/ECP
     username = netid
     factor = push
   actual (4 lines):
     [test]
     ecp_endpoint_url = https://shibboleth.illinois.edu/idp/profile/SAML2/SOAP/ECP
     username = netid
     factor = push
   --
   
 ✗ Configure empty default profile
   (from function `assert_equal' in file tests/../test_helper/bats-assert/src/assert_equal.bash, line 40,
    from function `test_default_profile' in file tests/login_configure.bats, line 43,
    in test file tests/login_configure.bats, line 57)
     `test_default_profile' failed
   Configured for AWSCLI V1.
   
   -- values do not equal --
   expected (4 lines):
     [default]
     ecp_endpoint_url = https://shibboleth.illinois.edu/idp/profile/SAML2/SOAP/ECP
     username = netid
     factor = push
   actual (4 lines):
     [test]
     ecp_endpoint_url = https://shibboleth.illinois.edu/idp/profile/SAML2/SOAP/ECP
     username = netid
     factor = push
   --
   
 ✗ Configure nonempty default profile
   (from function `assert_output' in file tests/../test_helper/bats-assert/src/assert_output.bash, line 194,
    in test file tests/login_configure.bats, line 80)
     `Overwrite profile 'default' to enable login? ECP Endpoint URL [None]: Username [None]: Enable Keyring [False]: Duo Factor [None]: Role ARN [None]: "' failed
   Configured for AWSCLI V1.
   
   -- output differs --
   expected (2 lines):
     WARNING: Profile 'default' contains credentials.
     Overwrite profile 'default' to enable login? ECP Endpoint URL [None]: Username [None]: Enable Keyring [False]: Duo Factor [None]: Role ARN [None]:
   actual (1 lines):
     ECP Endpoint URL [None]: Username [None]: Enable Keyring [False]: Duo Factor [None]: Role ARN [None]:
   --
   
 ✗ Do not configure nonempty default profile
   (from function `assert_output' in file tests/../test_helper/bats-assert/src/assert_output.bash, line 194,
    in test file tests/login_configure.bats, line 90)
     `Overwrite profile 'default' to enable login? "' failed
   Configured for AWSCLI V1.
   
   -- output differs --
   expected (2 lines):
     WARNING: Profile 'default' contains credentials.
     Overwrite profile 'default' to enable login?
   actual (1 lines):
     ECP Endpoint URL [None]: Username [None]: EOF when reading a line
   --
   
plugin.bats
 ✓ Ensure plugin does not require external modules
profiles.bats
 ✗ Login and logout with two profiles
   (from function `assert_equal' in file tests/../test_helper/bats-assert/src/assert_equal.bash, line 40,
    from function `login_test' in file tests/profiles.bats, line 53,
    in test file tests/profiles.bats, line 66)
     `login_test # Run login and shared tests' failed
   Configured for AWSCLI V1.
   ECP Endpoint URL [None]: Username [None]: Enable Keyring [False]: Duo Factor [None]: Role ARN [None]: ECP Endpoint URL [https://shibboleth.illinois.edu/idp/profile/SAML2/SOAP/ECP]: Username [netid]: Enable Keyring [False]: Duo Factor [push]: Role ARN [None]:
   -- values do not equal --
   expected (4 lines):
     [default]
     credential_process = aws-login --profile default
     [test]
     credential_process = aws-login --profile test
   actual (2 lines):
     [test]
     credential_process = aws-login --profile test
   --
   
 ✗ Logout --all
   (from function `assert_equal' in file tests/../test_helper/bats-assert/src/assert_equal.bash, line 40,
    from function `login_test' in file tests/profiles.bats, line 53,
    in test file tests/profiles.bats, line 89)
     `login_test # Run login and shared tests' failed
   Configured for AWSCLI V1.
   ECP Endpoint URL [None]: Username [None]: Enable Keyring [False]: Duo Factor [None]: Role ARN [None]: ECP Endpoint URL [https://shibboleth.illinois.edu/idp/profile/SAML2/SOAP/ECP]: Username [netid]: Enable Keyring [False]: Duo Factor [push]: Role ARN [None]:
   -- values do not equal --
   expected (4 lines):
     [default]
     credential_process = aws-login --profile default
     [test]
     credential_process = aws-login --profile test
   actual (2 lines):
     [test]
     credential_process = aws-login --profile test
   --
   

15 tests, 9 failures

make[1]: *** [tests] Error 1
make: *** [integration-tests] Error 2
```

--- a/src/integration_tests/tests/clean.bash
+++ b/src/integration_tests/tests/clean.bash
@@ -9,6 +9,7 @@ setup() {
     export AWS_CONFIG_FILE="$TEST_TEMP_DIR/.aws/config"
     export AWS_SHARED_CREDENTIALS_FILE="$TEST_TEMP_DIR/.aws/credentials"
     export AWSCLI_LOGIN_ROOT="$TEST_TEMP_DIR"
+    unset AWS_PROFILE  # If set can cause tests to fail (See #236)
 }
 
 teardown() {

